### DISCUSSION TITLE
Implementing a PATCH/PUT Endpoint for Nested Data in Mockoon

### DISCUSSION THREAD

**Discussion by jcbenck** [mockoon/mockoon#1245] [Help]
I have a nested data structure representing institutions and their associated requests. I'm using Mockoon for mocking the API. Below is an overview of my data setup:

Institutions DataBucket:
```
[
  {{# repeat (queryParamRaw 'total')}}
    {
      "id": "{{ faker 'string.uuid' }}",
      "name": "{{ faker 'company.name' }}"
    }
  {{/ repeat}}
]
```

Requests DataBucket:
```
[
  {{#each (dataRaw 'Institutions')}}
  {
      "institution": "{{this.id}}",
      "requests": [
          {{#repeat (queryParamRaw 'total')}}
          {
              "id": "{{faker 'string.uuid'}}",
              "profileId": "{{faker 'string.uuid'}}",
              "manuscriptId": "{{faker 'string.uuid'}}",
              "transactionId": "{{faker 'string.uuid'}}",
              ...
              ]
          }{{#unless @last}},{{/unless}}
          {{/repeat}}
      ]
  }{{#unless @last}},{{/unless}}
  {{/each}}
]
```

Sample Response from a GET by ID Endpoint (CRUD Route):
```
{
    "institution": "0f6b69ee-e806-4a69-8437-e012fea8b36d",
    "requests": [
        {
            "id": "f6aa7519-2bba-43af-be50-5250dfaadc7b",
            "profileId": "2bfb5bcf-f4c4-467e-a83f-f1293b532cf6",
            "manuscriptId": "6d4b83cb-eb1d-4959-867b-2183ef42eafb",
            "transactionId": "7ec6c1b5-bc87-4180-bde4-aa2cedd25ba4",
            ...
        },
            "statusCode": 2,
            "funderName": [
                "Armstrong, Kutch and Kling"
            ]
        }
    ]
}
```
I need to implement a PATCH or PUT endpoint that can handle multiple values for modification keys, such as `institutionId `and `requestId`, allowing me to modify information from a single field inside a request. My challenge is structuring this endpoint to efficiently update specific nested fields within this data structure.

I was trying an approach similar to this:
![image](https://github.com/mockoon/mockoon/assets/91486686/9b8b0b77-a8fb-4908-b4f0-68808e8c3e2d)

I was expecting to make a call like this
/requests/:institutionId/:requestId

Has anyone faced a similar challenge or can provide insights into how to structure this endpoint? Any suggestions or examples would be greatly appreciated!

**Comment by 255kb** [comment]
There is no support yet for nested resources and the field you highlighted does not support the dot notation to use a deep property as the id. 
I guess for now, what you are trying to achieve is not possible, unless your requests data are more flat (the requests array at the root level).
